src/App.js

```diff
@@ -1,13 +1,97 @@
 import React from 'react';
+import TodoForm from './components/TodoComponents/TodoForm'
+import TodoList from './components/TodoComponents/TodoList'
+import './components/TodoComponents/Todo.css';
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.testRef = React.createRef();
+    this.state = {
+      currentInputValue: '',
+      todos : []
+      };
+      window.onbeforeunload = this.saveData;
```

Comment:
This is nice, you could also do it with React lifecycle methods or change it whenever you update with new todos.